framework/base/src/types/interaction/tx.rs

```diff
@@ -418,3 +419,105 @@ where
         }
     }
 }
+
+impl<Env, Payment, Gas, Data> Tx<Env, (), (), Payment, Gas, Data, ()>
+where
+    Env: TxEnv,
+    Payment: TxPayment<Env>,
+    Gas: TxGas<Env>,
+    Data: TxData<Env>,
+{
+    pub fn deploy(
+        self,
+        code: ManagedBuffer<Env::Api>,
+        metadata: CodeMetadata,
+        arg_buffer: ManagedArgBuffer<Env::Api>,
+    ) -> Tx<Env, (), (), Payment, Gas, TxDataDeploy<Env>, ()> {
+        Tx {
+            env: self.env,
+            from: self.from,
+            to: self.to,
+            payment: self.payment, //this should be egld payment only
+            gas: self.gas,
+            data: TxDataDeploy::new(code, metadata, arg_buffer),
+            callback: self.callback,
+        }
+    }
+}
+
+impl<Env, Gas> Tx<Env, (), (), EgldPayment<Env::Api>, Gas, TxDataDeploy<Env>, ()>
```

Comment:
Won't work without payment.

Make a trait like "TxEgldOnlyPayment` that is implemented for EgldPayment and (). Doesn't need methods, you can use normalize() to get the EGLD value.